recipes/default.rb

```diff
@@ -51,17 +51,53 @@
   action :nothing
 end
 
-template '/etc/init/chef-guard.conf' do
-  source 'chef-guard.upstart.erb'
-  backup false
-  variables(
-    :recipe_file => (__FILE__).to_s.split("cookbooks/")[1],
-    :template_file => source.to_s,
-    :basedir => node['chef-guard']['install_dir']
-  )
-  notifies :restart, 'service[chef-guard]'
+# Set init style for supported platforms
+case node['platform_family']
+when 'rhel'
+  # CentOS/RHEL 7 and up
+  if node['platform_version'].to_i >= 7
+    INIT_SYSTEM = 'systemd'
+  # CentOS/RHEL 6 (not supporting 5)
+  else
+    INIT_SYSTEM = 'upstart'
+  end
+when 'debian'
+  # For ubuntu 14.10 and below
+  if node['platform'] == 'ubuntu' && node['platform_version'].to_i <= 14.10
+    INIT_SYSTEM = 'upstart'
+  # For all other ubuntu/debian versions (ubuntu 14.10 and up, debian 8 and up)
+  else
+    INIT_SYSTEM = 'systemd'
+  end
 end
```

Comment:
Think this logic can be removed and use the ohai information `node['init_package']` in the case statement here below.